src/Plutip/UtxoDistribution.purs

```diff
@@ -94,6 +94,18 @@ instance UtxoDistribution InitialUTxOsWithStakeKey KeyWallet where
         tail
   keyWallets _ wallet = [ wallet ]
 
+instance UtxoDistribution (Array InitialUTxOs) (Array KeyWallet) where
+  encodeDistribution amounts = amounts
+  decodeWallets d p = decodeWalletsDefault d p
```

Comment:
Avoid using `decodeWalletsDefault`, it has `guard $ Array.null remainingPKeys` which causes `Nothing` as a result

Reply:
thank you, I've changed the implementation and the test is passing now :)